code/game/objects/items/storage/belt.dm

```diff
@@ -15,6 +15,34 @@
 	cant_hold = list(/obj/item/weapon/throwing_knife)
 	///TRUE Means that it closes a flap over its contents, and therefore update_icon should lift that flap when opened. If it doesn't have _half and _full iconstates, this doesn't matter either way.
 	var/flap = TRUE
+	var/holster_gun = FALSE
+
+/obj/item/storage/belt/gun/dump_into(obj/item/storage/M, mob/user)
+	for(var/slot in holster_slots)
+		if(!holster_slots[slot]["gun"]) //Open holster.
+			holster_gun = FALSE
+			break
+		if(holster_slots[slot]["gun"]) //Gun in holster.
+			holster_gun = TRUE
+			break
+
+	if(holster_gun == FALSE && contents.len >= (storage_slots-1))
+		to_chat(user, SPAN_WARNING("[src] is full."))
+		return FALSE
+	..()
+
+/obj/item/storage/belt/gun/handle_item_insertion(obj/item/W, prevent_warning = 0, mob/user)
+	for(var/slot in holster_slots)
```

Comment:
Also is there any reason why you're doing it this way vs just checking the list of holstered guns like in the original suggestion on discord? Did you find the flaregun not get added to that list?

Reply:
Like in my original snippet, why not look at the list `holstered_guns` rather than `holster_slots`. `holster_slots` is for the inventory representation. `holstered_guns` is just a list of guns in the gun belt. If the issue is just that the flaregun doesn't get counted as a gun to be entered into holstered_guns, then thats likely more the problem (e.g. `/obj/item/storage/belt/gun/_item_insertion(obj/item/W, prevent_warning = 0)` needs to have an override for /obj/item/storage/belt/gun/**flare_gun**/_item_insertion that'll treat a flaregun as a gun to be inserted in that list. But anyways, I'd just try it out and see if that simplifies things for you or not.

Reply:
your right i must have lost the holstered_guns sugestion while trying stuff around.
i see flaregun isn't consider a gun.... that could cause other issue in other part of the code?
i will change the insertion item to only be for flargun as it's it's only purpose...
thanks.